### PR TITLE
24 separating workload language from the language used for search

### PR DESCRIPTION
**Pull request by seyhani**

### CODE REVIEW

File: lib/spec_factory.hpp [hunk 1]
```diff
@@ -78,7 +78,12 @@ class SpecFactory {
     //**** COMP ****/
     op_t random_op();
 
+    typedef WlSpec* (SpecFactory::*SpecGeneratorFuncPtr)();
+    static vector<SpecGeneratorFuncPtr> spec_generators;
+
 private:
+    static void initializeSpecs();
```
Comment (seyhani) [inline lib/spec_factory.hpp:85]: I think now we can make it non-static 🤔

File: src/spec_factory.cpp [hunk 2]
```diff
@@ -88,7 +100,10 @@ void SpecFactory::pick_neighbors(TimedSpec& spec, vector<TimedSpec>& neighbors)
 //************************************* WlSpec *************************************//
 
 WlSpec* SpecFactory::random_wl_spec() {
-    return new Comp(random_comp());
+    std::mt19937 gen;
```
Comment (seyhani) [inline src/spec_factory.cpp:103]: ``
```suggestion
    mt19937& gen = dists->get_gen();
```

File: src/spec_factory.cpp [hunk 3]
```diff
@@ -88,7 +100,10 @@ void SpecFactory::pick_neighbors(TimedSpec& spec, vector<TimedSpec>& neighbors)
...
+    std::uniform_int_distribution<> dist(0, spec_generators.size() - 1);
```
Comment (seyhani) [inline src/spec_factory.cpp:104]:
```suggestion
    uniform_int_distribution<> dist(0, spec_generators.size() - 1);
```

File: lib/spec_factory.hpp (hunk 1 above)
Comment (lucedes27) [reply lib/spec_factory.hpp]: [Update spec_factory.hpp](https://github.com/all-things-networking/fperf/pull/28/commits/f99d01efde44d782b7a3de03cf89638cfb24a3b7)

File: lib/spec_factory.hpp [hunk 4]
```diff
@@ -78,7 +78,12 @@ class SpecFactory {
     //**** COMP ****/
     op_t random_op();
 
+    typedef WlSpec* (SpecFactory::*SpecGeneratorFuncPtr)();
+    static vector<SpecGeneratorFuncPtr> spec_generators;
```
Comment (seyhani) [inline lib/spec_factory.hpp:82]:
```suggestion
   vector<SpecGeneratorFuncPtr> spec_generators;
```

**Pull request merged**